client/src/components/VideoForm.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from "react";
+import validUrl from "valid-url";
+
+function VideoForm({ videoData, setVideoData, setFetchData }) {
+  const [errorMessage, setErrorMessage] = useState("");
+
+  function addVideoHandler(event) {
+    event.preventDefault();
+    function validateUrl(urlObject) {
+      return (
+        validUrl.isUri(urlObject) &&
+        (urlObject.startsWith("https://www.youtube.com") ||
+          urlObject.startsWith("https://youtu.be") ||
+          urlObject.startsWith("https://m.youtube.com") ||
+          urlObject.startsWith("https://youtube.com/"))
+      );
+    }
+
+    const formTitle = event.target.form.title.value;
+    const formUrl = event.target.form.url.value;
+
+    const cleanedUrl = formUrl.includes("&") ? formUrl.split("&")[0] : formUrl;
+
+    if (validateUrl(formUrl) && formTitle !== "") {
+      const newData = {
+        id: parseInt(Math.random() * 1000),
```

Comment:
1000 is a pretty small number of IDs, and is likely to have collisions (multiple videos with the same ID) with only a few videos. (See https://en.wikipedia.org/wiki/Birthday_problem for some interesting information on the probabilities here!)

Also, the server should probably be picking the IDs, not the client, as the server is the thing that knows what IDs are already used.